indico/modules/events/registration/fields/simple.py

```diff
@@ -571,30 +572,62 @@ def process_form_data(self, registration, value, old_data=None, billable_items_l
         # always store no-option as empty dict
         if value is None:
             value = {}
+
+        return_value = {}
+
+        if old_data:
+            old_choices_mapping = {x['id']: x for x in old_data.field_data.versioned_data['choices']}
+            new_choices_mapping = {x['id']: x for x in self.form_item.versioned_data['choices']}
+            revoked_selected_choice_ids = []
+            selected_new_option_exists = False
+            selected_option_ids = value.keys()
+
+            for selected_option_id in selected_option_ids:
+                # if an item from old_data's version was already chosen but doesn't exist anymore
+                # TODO: or has a new price
+                if selected_option_id not in new_choices_mapping.keys():
+                    revoked_selected_choice_ids.append(selected_option_id)
+                # the user also selected a new item that doesn't exist in the current version yet
+                elif selected_option_id not in old_choices_mapping.keys():
+                    selected_new_option_exists = True
+            if selected_new_option_exists and revoked_selected_choice_ids:
+                # Creating a new data version
+                new_data_version = deepcopy(self.form_item.versioned_data)
```

Comment:
I think `self.form_item.current_version.versioned_data` would be better since `form_item.versioned_data` wraps the data in an `ImmutableDict` (which is preserved in deepcopy). The only reason why it still works here is that `choices` inside that dict is not immutable (which I'd consider a but in the `versioned_data` property - the idea of returning an immutable dict there is exactly to disallow modification when using the shortcut instead of explicitly going through the version)